### ISSUE TITLE
[Bug]: Clicking on "Upen Updater" leads to a 404 error.

### ISSUE THREAD

**Issue by ccpf** [nextcloud/server#42206]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I have v.25.0.3 (note that I am unable to update, I also cannot manually update as I am on a hosted server without sudo access). When I click on "Open Updater" I receive a 404 error.

### Steps to reproduce

1. Go to "administration -> overview"
2. Click on "Open Updater"
3. I get a 404 error page.

### Expected behavior

Update the installation

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

26

### Operating system

Other

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
As it is a hosted webserver I obviously don't have sudo access.
```

### List of activated Apps

```shell
as above
```

### Nextcloud Signing status

```shell
"No errors have been found"
```

### Nextcloud Logs

```shell
There is no entry with today's date. In updater.log, there are these entries:

2023-12-13T12:05:53+0100 VRVLoKnLGh [info] request to updater
2023-12-13T12:05:53+0100 VRVLoKnLGh [info] currentStep()
2023-12-13T12:05:53+0100 VRVLoKnLGh [info] show HTML page
2023-12-13T12:05:53+0100 VRVLoKnLGh [info] current version: 25.0.3 build time: 2023-01-17T09:08:33+00:00 494a0c1073b460d4d2a5c03ff3567fc388b91e36
2023-12-13T12:05:57+0100 v1rjkBus3u [info] request to updater
2023-12-13T12:05:57+0100 v1rjkBus3u [info] currentStep()
2023-12-13T12:05:57+0100 v1rjkBus3u [info] show HTML page
2023-12-13T12:05:57+0100 v1rjkBus3u [info] current version: 25.0.3 build time: 2023-01-17T09:08:33+00:00 494a0c1073b460d4d2a5c03ff3567fc388b91e36
2023-12-13T12:08:09+0100 HksW24NoCd [info] request to updater
2023-12-13T12:08:09+0100 HksW24NoCd [info] currentStep()
2023-12-13T12:08:09+0100 HksW24NoCd [info] show HTML page
2023-12-13T12:08:09+0100 HksW24NoCd [info] current version: 25.0.3 build time: 2023-01-17T09:08:33+00:00 494a0c1073b460d4d2a5c03ff3567fc388b91e36
2023-12-13T12:11:01+0100 vUqGfXQrY7 [info] request to updater
2023-12-13T12:11:01+0100 vUqGfXQrY7 [info] currentStep()
2023-12-13T12:11:01+0100 vUqGfXQrY7 [info] show HTML page
2023-12-13T12:11:01+0100 vUqGfXQrY7 [info] current version: 25.0.3 build time: 2023-01-17T09:08:33+00:00 494a0c1073b460d4d2a5c03ff3567fc388b91e36
2023-12-13T12:11:27+0100 qUrCDXNWNh [info] request to updater
2023-12-13T12:11:27+0100 qUrCDXNWNh [info] currentStep()
2023-12-13T12:11:27+0100 qUrCDXNWNh [info] show HTML page
2023-12-13T12:11:27+0100 qUrCDXNWNh [info] current version: 25.0.3 build time: 2023-01-17T09:08:33+00:00 494a0c1073b460d4d2a5c03ff3567fc388b91e36
2023-12-13T12:12:18+0100 ApaAewgbVH [info] request to updater
2023-12-13T12:12:18+0100 ApaAewgbVH [info] currentStep()
2023-12-13T12:12:18+0100 ApaAewgbVH [info] show HTML page
2023-12-13T12:12:18+0100 ApaAewgbVH [info] current version: 25.0.3 build time: 2023-01-17T09:08:33+00:00 494a0c1073b460d4d2a5c03ff3567fc388b91e36
```

### Additional info

I have the following messages on the admin page:

`Your web server is not properly set up to resolve "/datacloud/ocm-provider/". This is most likely related to a web server configuration that was not updated to deliver this folder directly. Please compare your configuration against the shipped rewrite rules in ".htaccess" for Apache or the provided one in the documentation for Nginx at it's [documentation page ↗](https://docs.nextcloud.com/server/25/go.php?to=admin-nginx). On Nginx those are typically the lines starting with "location ~" that need an update.`

and

`MySQL is used as database but does not support 4-byte characters. To be able to handle 4-byte characters (like emojis) without issues in filenames or comments for example it is recommended to enable the 4-byte support in MySQL. For further details read [the documentation page about this ↗](https://docs.nextcloud.com/server/25/go.php?to=admin-mysql-utf8mb4).`

Regarding the first one, I did not change htaccess and am not on Nginx. Regarding the 2nd one, that should not be relevant.

**Comment by ccpf** [comment]
> > I also cannot manually update as I am on a hosted server without sudo access).
> 
> A manual update does _not_ require sudo access. Merely the ability to access your files remotely via sftp/ssh/ftp or similar.
> 
> Please read the Admin Manual _Maintenance_ section at https://docs.nextcloud.com for details.
> 
> This looks to be an installation/setup matter rather than a bug report so please post at the Nextcloud Help Forum for any further assistance: https://help.nextcloud.com

According to this page I need to restart my webserver in step 13 (which I can't) and need sudo acccess for step 14 (which I don't have).

**Comment by joshtrichards** [comment]
The use of sudo merely is an example. It's not necessary unless you used it originally during installation. The only access necessary is whatever access you had when you installed Nextcloud in your environment.

**Comment by ccpf** [comment]
I installed it via Softacolous which is part of my cPanel hosted server package. However, when I try to update it via Softaculous I get the error that it cannot upgrade from v25.x to 27, and I need to upgrade to 26 first. Do you maybe know how this can be done? I seems that I cannot choose the version that Softaculous updates to, so I would either need to the the update to 26 or 27 manually.

**Comment by ccpf** [comment]
... and if that should not be possible, say I manually install v.27 into a different installation directory while keeping the same directory (the latter is outside my installation directory). Would this work? I.e., if I just copied over the config file and changed url in my desktop application to the new installation directory, would this work? Would the new installation update the database to the new format?

**Comment by ccpf** [comment]
I managed to resolve it. In case others have the same problem, here the solution:

1. If you are on v25, download the last v26 file and place it into an empty directory on your hosted server and unpack it.
2. Follow the step from here https://docs.nextcloud.com/server/latest/admin_manual/maintenance/manual_upgrade.html (leave out the sudo part of the update command and also omit the part on the file permissions entirely as they are complete nonsense, it essentially makes your nextcloud directory unreadable if you access it through a browser, I had to manually change all permissions again).
3. Then go to Softaculous and try to update. You will get a message at the top saying that Softaculous thinks it is version 25.X but it actually is v26.X. Click the message to update the Softaculous records.
4. Now Softaculous will perform the update from v26 to v27.
